filters/filters.go

```diff
@@ -151,6 +151,15 @@ type Filter interface {
 	Response(FilterContext)
 }
 
+// FilterCloser are Filters that need to cleanup resources after
+// filter teminiation. For example Filters, that create a goroutine
+// for some reason need to cleanup their goroutine or they would leak
+// goroutines.
+type FilterCloser interface {
+	Filter
+	Close()
+}
```

Comment:
We implement goroutine cleanup via admissionControlPost

Maybe in your test you could utilize it and e.g. update routing table to an empty value just before test ends such that post-processor closes all instances.

Alternatively we may call cleanup manually on the post-postprocessor like it is done here https://github.com/zalando/skipper/blob/e57538ee7028f79baa0a285db268c4e7c3cf346e/filters/auth/main_test.go#L11-L17

Reply:
This is related to the long known but hard problem of filter resource cleanup https://github.com/zalando/skipper/issues/202

Reply:
How can you access the PostProcessor from the test?

Reply:
I found a way to fix all the things. It did not require a TestMain hack.